src/_pytest/assertion/util.py

```diff
@@ -169,23 +190,56 @@ def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[
 
 
 def _compare_eq_any(left: Any, right: Any, verbose: int = 0) -> List[str]:
-    explanation = []  # type: List[str]
+    """
```

Comment:
`_compare_eq_any` is an internal function which the reader is kind of expected to understand what it does on their own. I think a one sentence docstring

> Generic comparison function, dispatches to other comparison functions based on the inputs' type.

would be nice, however the rest is mostly redundant with the function signature, so I'd leave it out.